### PR TITLE
added python support

### LINKED ISSUE #12: Add python support

### PR DESCRIPTION
**Pull request by cocoa-xu**
Hi, this PR adds python support for ruapu. It should close #12.  ;)

### CODE REVIEW

File: ruapu-py.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{
+    PyModuleDef_HEAD_INIT,
+    "ruapu",
+    "ruapu module",
+    -1,
+    ruapu_methods
+};
+
+PyMODINIT_FUNC PyInit_ruapu(void)
+{
```
Comment (nihui) [inline ruapu-py.c:42]: How about putting the initialization process here?
Initialize when `import ruapu`, eliminating the need to call init again.

File: README.md [hunk 2]
```diff
@@ -77,6 +79,27 @@ xop = 0
...
+$ python3
+>>> import ruapu
+>>> ruapu.init()
+>>> ruapu.supports("avx2")
+True
+>>> ruapu.supports(isa="avx2")
+True
+...
+```
+
```
Comment (nihui) [inline README.md:102]: Use table to form 2 columns, just like C

File: ruapu-py.c (hunk 1 above)
Comment (cocoa-xu) [reply ruapu-py.c:44]: Possible but perhaps we can further delay the initialization process at the first call of `ruapu.supports`?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -31,6 +31,10 @@ jobs:
       run: |
         clang main.c -o ruapu-clang
         ./ruapu-clang
+    - name: build-test-python
+      run: |
+        pip3 install .
+        python3 -c 'import ruapu; print(ruapu.supports("neon")); print(ruapu.supports(isa="avx"))'
```
Comment (nihui) [inline .github/workflows/ci.yml:37]: ci log shows that an UNKNOWN package was built
try upgrading pip or setuptools ?

```
Building wheels for collected packages: UNKNOWN
  Building wheel for UNKNOWN (pyproject.toml): started
  Building wheel for UNKNOWN (pyproject.toml): finished with status 'done'
  Created wheel for UNKNOWN: filename=UNKNOWN-0.0.0-cp310-cp310-linux_x86_64.whl size=13600 sha256=8084b088ef4675e5144eb1b6fe5e592eebf3dc1744ed0a511e9613562394fd44
  Stored in directory: /tmp/pip-ephem-wheel-cache-smr2q52j/wheels/6e/d5/b4/854280212ff45b045af166a62e2eeac73e7a772e78fd439104
Successfully built UNKNOWN
Installing collected packages: UNKNOWN
Successfully installed UNKNOWN-0.0.0
```

Comment (cocoa-xu) [reply .github/workflows/ci.yml:39]: Updating setuptools seems to solve it. ;)

**Review by nihui — APPROVED** [review]

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**